client/my-sites/stats/subscribers-section/index.tsx

```diff
@@ -87,11 +113,19 @@ export default function SubscribersSection( {
 						</small>
 					) }
 				</h1>
-				<div className="subscribers-section-duration-control-with-legend">
-					<StatsPeriodHeader>
-						<Intervals selected={ period } pathTemplate={ pathTemplate } compact={ true } />
-					</StatsPeriodHeader>
-					<div className="subscribers-section-legend" ref={ legendRef }></div>
+				<div className="subscribers-section-heading__chart-controls">
+					<SubscribersNavigationArrows
+						date={ queryDate }
+						period={ period }
+						quantity={ quantity }
+						onDateChange={ handleDateChange }
+					/>
+					<div className="subscribers-section-duration-control-with-legend">
+						<StatsPeriodHeader>
+							<Intervals selected={ period } pathTemplate={ pathTemplate } compact={ true } />
```

Comment:
I noticed that `queryDate` persists between period changes. What do you think about resetting the `queryDate` on period changes?

Reply:
Good idea. Resetting it to the current date should make the navigation more predictable.